consensus/src/pipeline/buffer_manager.rs

```diff
@@ -765,6 +792,12 @@ impl BufferManager {
             .set(pending_aggregated as i64);
     }
 
+    fn need_backpressure(&self) -> bool {
```

Comment:
@igor-aptos  fyi I added backpressure here.. so that we don't switch to state sync with a pre-committed version > target LI version